.github/workflows/test.yml

```diff
@@ -42,63 +44,61 @@ jobs:
 
     # Test end-to-end by uploading two artifacts and then downloading them
     # Once upload-artifact v2 is out of preview, switch over
+    - name: Set artifact file contents
+      run: |
+        echo "::set-env name=non-gzip-artifact-content::hello"
+        echo "::set-env name=gzip-artifact-content::Some_large_amount_of_text_that_has_a_compression_ratio_that_is_greater_than_100%._If_the_compression_ratio_is_greater_than_100%_gzip_is_used_to_upload_the_file."
+        
     - name: Create artifacts
       run: |
         mkdir -p path/to/artifact-A
         mkdir -p path/to/artifact-B
-        echo "Lorem ipsum dolor sit amet" > path/to/artifact-A/file-A.txt
-        echo "Hello world from file B" > path/to/artifact-B/file-B.txt
+        echo ${{ env.non-gzip-artifact-content }} > path/to/artifact-A/file-A.txt
+        echo ${{ env.gzip-artifact-content }} > path/to/artifact-B/file-B.txt
     
     - name: Upload artifact A
-      uses: actions/upload-artifact@v1
+      uses: actions/upload-artifact@v2
       with:
         name: 'Artifact-A'
         path: path/to/artifact-A
 
     - name: Upload artifact B
-      uses: actions/upload-artifact@v1
+      uses: actions/upload-artifact@v2
       with:
         name: 'Artifact-B'
         path: path/to/artifact-B
 
-    # Test downloading a single artifact
-    - name: Download artifact A
+    - name: Download artifact A with no extra folder (default)
+      uses: ./
+      with:
+        name: 'Artifact-A'
+        path: download/with/artifact-folder/false
+
+    - name: Verify successful download with no extra folder
+      shell: bash
+      run: |
+        scripts/test-artifact-file.sh "download/with/artifact-folder/false/file-A.txt" "${{ env.non-gzip-artifact-content }}"
```

Comment:
The tests here are largely identical to those found in [actions/artifact](https://github.com/actions/toolkit/blob/master/.github/workflows/artifact-tests.yml). In the future I would like to get rid of env variables and do a binary comparison across the files, but for the moment, I'm sticking to what works